src/data/finsupp/pointwise.lean

```diff
@@ -70,4 +70,34 @@ instance [non_unital_semiring β] : non_unital_semiring (α →₀ β) :=
 { ..(infer_instance : semigroup (α →₀ β)),
   ..(infer_instance : non_unital_non_assoc_semiring (α →₀ β)) }
 
+-- TODO can this be generalized in the direction of `pi.has_scalar'`
+-- (i.e. dependent functions and finsupps)
+-- TODO in theory this could be generalised, we only really need `smul_zero` for the definition
+/-- The pointwise multiplicative action of functions on finitely supported functions -/
+instance pointwise_distrib_mul_action [monoid β] [add_monoid γ]
+  [distrib_mul_action β γ] : distrib_mul_action (α → β) (α →₀ γ) :=
```

Comment:
When `γ = α → β`, is the diamond formed by this instance defeq? If so, it would be nice to add an `example` below to prove it.

Reply:
Sorry, I mean set `γ := (α → β)`, and then the diamond is by the existing distrib_mul_action instance

Reply:
Ah I see thanks, well unfortunately it seems it's not defeq as this fails
```

example [ring β] [add_monoid γ]
  [distrib_mul_action β γ]: @finsupp.pointwise_distrib_mul_action α β (α → β) _ _ _
  = finsupp.distrib_mul_action _ (α → β) := rfl
```
I can't see a way to fix this, I'm not even sure if it's mathematically fixable lol, maybe a better approach is just to be more conservative and only define this action when `beta = gamma` is a ring, this was my original use case in fact, that would avoid any overlap in these actions it seems

Reply:
Yeah, it's mathematically false:
```lean
attribute [ext] distrib_mul_action mul_action has_scalar

example [ring β] [add_monoid γ]
  [distrib_mul_action β γ]: @finsupp.pointwise_distrib_mul_action α β (α → β) _ _ _
  = finsupp.distrib_mul_action _ (α → β) :=
  begin
    ext fab faab a₁ a₂,
    dsimp,
    sorry -- fab a₁ * ⇑faab a₁ a₂ = fab a₂ * ⇑faab a₁ a₂
  end
  ```

Reply:
It turns out that `pi.has_scalar'`, which this PR emulates, [is no better](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Diamond.20in.20docs.23pi.2Ehas_scalar'/near/266008253)...

Reply:
Lol, nice! I'd also be happy to namespace this one then, or just restrict to beta = gamma here (do you see any potential for diamonds there?)

Reply:
Another option for now would  just be adding a scary warning to the docstring like:
```
**Warning**: This instance can form diamonds with `finsupp.distrib_mul_action` when `γ` is `α → β`
```

> I'd also be happy to namespace this one then, or just restrict to beta = gamma here

This is probably safe too.

Reply:
Ok I went for the less scary option and specialized this to ring